### PR TITLE
rpc: user regexp replace string match scheme

### PR DESCRIPTION
**Pull request by hello2dj**
Signed-off-by: dengjie <dengjie.hello2dj@bytedance.com>

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,6 +18,7 @@ integration-tests = []
 name = "tikv_client"
 
 [dependencies]
+regex = "1"
```
Comment (ice1000) [inline Cargo.toml]: Could you please use a more generic version? Like `^1` (or probably something better)?

**Comment by ice1000** [comment]
Please run `cargo fmt`

**Comment by ice1000** [comment]
Missing signoffs, LGTM

**Comment by hello2dj** [comment]
thank you

File: Cargo.toml (hunk 1 above)
Comment (nrc) [reply Cargo.toml]: I think `1` is fine - it means any version back compat with 1 (i.e., the same as `0.1` for the other deps)

File: src/rpc/security.rs [hunk 2]
```diff
@@ -65,9 +72,10 @@ impl SecurityManager {
         Factory: FnOnce(Channel) -> Client,
     {
         info!("connect to rpc server at endpoint: {:?}", addr);
-        let addr = addr
-            .trim_start_matches("http://")
-            .trim_start_matches("https://");
+
+        let cow_addr = SCHEME_REG.replace(addr, "");
+        let addr: &str = cow_addr.borrow();
```
Comment (nrc) [inline src/rpc/security.rs]: Could you use `&addr` below and avoid the explicit call to `borrow`?

Comment (hello2dj) [reply src/rpc/security.rs]: done

**Comment by nrc** [comment]
CI is failing due to needing https://github.com/tikv/client-rust/pull/60

**Pull request merged**